common/lib/dependabot/pull_request_updater/azure.rb

```diff
@@ -27,6 +27,21 @@ def update
         update_source_branch
       end
 
+      def update_pr_elements(elements = {})
```

Comment:
What's the idea behind updating certain elements sometimes vs the entire PR other times?

A concern is that we now no longer follow the same API for different updaters.

I don't know if that's necessarily problematic, but up until now we've kept the interfaces the same between these different clients. At this point I think it's pretty unlikely that someone will agnostically open PRs against different providers, so deviating might be fine. @feelepxyz wdyt?

I've been thinking that it might even make sense to completely split off the Azure and GitLab clients from core and finding long-term maintainers for those, as we're probably not the best folks to maintain this code.

Reply:
@milind009 would be good to understand the use case for this and why the existing update method is not working for you? Slightly worried we're adding support for use-cases we don't understand and we end up with several slightly different ways to do the same thing.

Keen to figure if we can improve the existing update method instead or leave this as something that's done outside of dependabot if it's specific to your workflow.

Reply:
Hm just re-reading your PR description and sounds like we won't be able to easily fix the existing update method to do what you want. Looks like we only pass in `pr_name` in the creator for example and assuming you want to be able to update this?

I'm inclined to not merge this into core and encourage you to write a wrapper for this functionality in your dependabot setup as it seems specific to what you are doing.

Reply:
@feelepxyz Currently the update function that we have for the azure client is used force push new changes on to the PR. However, we have a requirement to set a PR to autocomplete (azure create PR API does have an option to set PR to autocomplete but unfortunately it does not work, hence we need to create a PR and then set it to autocomplete)/abandon existing PR for which we need to use the azure [update PR API](https://docs.microsoft.com/en-us/rest/api/azure/devops/git/pull%20requests/update?view=azure-devops-rest-6.0#:~:text=URI%20Parameters,-URI%20Parameters&text=The%20name%20of%20the%20Azure%20DevOps%20organization.&text=ID%20of%20the%20pull%20request%20to%20update.&text=The%20repository%20ID%20of%20the%20pull%20request's%20target%20branch.&text=Version%20of%20the%20API%20to%20use.). So I created to separate method which can be used for updating other PR aesthetics like title, description,etc.

Reply:
@milind009 is this something that could maybe live in the project that you use to run dependabot? It doesn't seem to rely on anything dependabot-specific?

Reply:
Cool makes sense. I'll add this in our project. Thanks for the prompt replies!